### PR TITLE
Long Control: Vibe Controller

### LINKED ISSUE #620: Longitudinal: Acceleration Personality
https://github.com/sunnypilot/sunnypilot/pull/519

### LINKED ISSUE #883: Longitudinal: Dynamic Personality
https://github.com/sunnypilot/sunnypilot/pull/882

### PR DESCRIPTION
**Pull request by rav4kumar**
This PR introduces the VibePersonalityController, a configurable personality system that separates acceleration behavior and braking/following distance logic into two independently tunable components:

AccelPersonality (_eco, normal, sport_): controls acceleration aggressiveness

LongPersonality (_relaxed, standard, aggressive_): governs braking and time gap behavior

This separation allows more flexible and human-like behavior tuning, as users can now mix personalities (e.g., eco acceleration with aggressive following).

**Key Features**

- ✅ Independent personalities for acceleration and following distance
- 🧠 Profiles stored in Params and updated dynamically at runtime
- 🚦 AccelPersonality affects maximum acceleration using velocity breakpoints and Hermite interpolation
- 🛑 LongPersonality controls minimum acceleration (braking) and following distance profiles
- 🔁 Personality cycling methods for easy toggling in UI or testing
- 📉 Precomputed Hermite slopes for efficient real-time interpolation
- ⚙️ Runtime toggle flags for full, partial, or disabled control

**Benefits**
 - Enables finer granularity and personalization in driving feel
- Facilitates more natural transitions between driving modes

## Summary by Sourcery

Introduce a configurable VibePersonalityController that decouples acceleration and following-distance logic into independent profiles, integrate it into the longitudinal control pipeline and UI, extend message schemas and params, and add unit tests for validation

New Features:
- Add VibePersonalityController to separate and tune acceleration (eco/normal/sport) and following distance/braking (relaxed/standard/aggressive) behaviors
- Add UI controls in offroad settings and onroad toggles for enabling and cycling acceleration and follow-distance personalities
- Extend longitudinal MPC and planner to query VibePersonalityController for acceleration limits and follow-distance multipliers with stock fallbacks
- Introduce AccelerationPersonality enum in cereal schema and add corresponding ParamKeys for persistent personality settings
- Add comprehensive unit tests covering initialization, parameter updates, toggles, and profile-based limit calculations

Enhancements:
- Store personality profiles in Params and update dynamically at runtime

Tests:
- Add unit tests for VibePersonalityController behavior, parameter throttling, cycling, and edge cases

### CODE REVIEW

File: common/params_keys.h [hunk 1]
```diff
@@ -123,6 +123,7 @@ inline static std::unordered_map<std::string, uint32_t> keys = {
     {"Version", PERSISTENT},
 
     // --- sunnypilot params --- //
+    {"AccelPersonality", PERSISTENT},
```
Comment (devtekve) [inline common/params_keys.h]: Make it backed up too

File: common/params_keys.h [hunk 2]
```diff
@@ -147,6 +148,9 @@ inline static std::unordered_map<std::string, uint32_t> keys = {
     {"QuickBootToggle", PERSISTENT | BACKUP},
     {"QuietMode", PERSISTENT | BACKUP},
     {"ShowAdvancedControls", PERSISTENT | BACKUP},
+    {"VibePersonalityEnabled", PERSISTENT},
```
Comment (devtekve) [inline common/params_keys.h]: Add backup to this and the others bellow

**Comment by sunnyhaibin** [comment]
Needs to be synced with the latest master and discuss how we want to split this PR in pieces.
